src/coreclr/jit/importer_vectorization.cpp

```diff
@@ -526,15 +526,15 @@ GenTree* Compiler::impExpandHalfConstEquals(GenTreeLclVar*   data,
         GenTreeColon* lenCheckColon = gtNewColonNode(TYP_INT, indirCmp, gtNewFalse());
 
         // For StartsWith we use GT_GE, e.g.: `x.Length >= 10`
-        lenCheckNode = gtNewQmarkNode(TYP_INT, gtNewOperNode(cmpOp, TYP_INT, lengthFld, elementsCount), lenCheckColon);
+        lenCheckNode = gtNewQmarkNode(TYP_BOOL, gtNewOperNode(cmpOp, TYP_INT, lengthFld, elementsCount), lenCheckColon);
```

Comment:
Is this a standard IR shape for qmarks? `TYP_BOOL` typed node is quite uncommon in the JIT IR.

Reply:
What ends up happening to the qmark when it is bool typed? How does this change the IR post qmark rationalization?

Reply:
I assume they're pretty legal till Lower, in this case we get rid of QMARK in morph always

Reply:
@jakobbotsch DIFF for the snippet in the description: https://www.diffchecker.com/QeDP5n5k (Main vs PR)
as you can see it creates an assertion (value is 0..255) for LCL we spill QMARK to. Then that assertion helps to get rid of CAST node.

Reply:
It would be interesting to see if there would be even better diffs by generalizing `IntegralRange::ForNode` slightly instead to handle qmarks (by union, I guess). Maybe commas too. From cursory glance the subrange assertion is coming from that.

This change is probably ok but it does look strange to have `TYP_BOOL` qmark with `TYP_INT` relop as 'then' branch, but I suppose we know that relops always leave the upper 31 bits cleared.

Reply:
@jakobbotsch hm, let's see what CI thinks about it

Reply:
Changed to be handled in `IntegralRange::ForNode` - no new diffs so far, so it's just a more verbose change now 🙂  maybe will help some future code (if we won't get rid of QMARK at that point yet)

Reply:
I wouldn't expect it for qmark, but maybe if you also handle commas 🙂

Reply:
> I wouldn't expect it for qmark, but maybe if you also handle commas 🙂

No new diffs (for libraries.pmi)